woocommerce/woocommerce-ios
# [Mobile Payments] Pair with single reader. Mark II: Yosemite

Closes #3715 

⚠️ This PR is against the feature branch implementing support for Card Present Payments, not against develop ⚠️

We continue building our prototype (for reference, p91TBi-4q4-p2), this time, by implementing logic in Yosemite to connect to a card reader. This is necessary to move on to the next step: processing a payment.

This PR starts where #3735 ends.

At this point we are not handling errors. We will tackle error handling in #3734 y #3741 

As usual, the code contains some comments that attempt to point out what we have left out of the implementation for now.

## Changes
* Add a new case to CardPresentReaderAction
* Typealias Yosemite.CardReader to Hardware.CardReader. Not strictly necessary for this PR, but, well, it's just one line 🤷 
* Add enough code to CardPresentPaymentStore to connect to a Reader, ignoring error handling
* Add a test to cover the happy path of the connection to a reader. Tests covering the potential errors will be implemented later

## How to test
These are not user facing changes. So a ✅ on CI and  👀 on the code should be enough. 

Alternatively:
* Checkout the branch, run `bundle exec pod install`. 
* Run the tests, specially those in CardPresentPaymentStoreTests.

Update release notes:
